packages/liquidation-sdk-viem/src/preLiquidation/logGetters.ts

```diff
@@ -0,0 +1,207 @@
+                    },
+                  ],
+                },
+              ],
+            },
+          }),
+        ),
+      )
+    ).flat();
+
+    return logs.map((log) => {
+      return {
+        marketId: log.args.id! as MarketId,
+        address: log.args.preLiquidation! as Address,
+        preLiquidationParams: formatPreLiquidationParams(
+          log.args.preLiquidationParams!,
+        ),
+      };
+    });
+  } catch (e) {
+    console.error(e);
+    return [];
+  }
+}
+
+export async function authorizationLogs<chain extends Chain = Chain>(
+  client: Client<Transport, chain>,
+  preLiquidation: PreLiquidation,
+) {
+  const chainId = ChainUtils.parseSupportedChainId(client.chain.id);
+
+  try {
+    const head = await getBlockNumber(client);
+
+    const intervals = sliceBlockInterval(
+      preLiquidationFactoryConfigs[chainId].startBlock,
+      BigInt(head),
+    );
+
+    const logs = (
+      await Promise.all(
+        intervals.map((interval) =>
+          getLogs(client, {
+            address: addresses[chainId].morpho,
+            fromBlock: interval.startBlock,
+            toBlock: interval.endBlock,
+            event: {
+              type: "event",
+              name: "SetAuthorization",
+              inputs: [
+                {
+                  name: "caller",
+                  type: "address",
+                  indexed: true,
+                  internalType: "address",
+                },
+                {
+                  name: "authorizer",
+                  type: "address",
+                  indexed: true,
+                  internalType: "address",
+                },
+                {
+                  name: "authorized",
+                  type: "address",
+                  indexed: true,
+                  internalType: "address",
+                },
+                {
+                  name: "newIsAuthorized",
+                  type: "bool",
+                  indexed: false,
+                  internalType: "bool",
+                },
+              ],
+              anonymous: false,
+            },
+            args: {
+              authorized: preLiquidation.address,
+            },
+          }),
+        ),
+      )
+    ).flat();
+
+    return logs
+      .filter((log) => log.args.newIsAuthorized! === true)
+      .map((log) => log.args.authorizer! as Address);
+  } catch (e) {
+    console.error(e);
+    return [];
+  }
+}
+
+function formatPreLiquidationParams(preLiquidiationParams: {
+  preLltv: bigint;
+  preLCF1: bigint;
+  preLCF2: bigint;
+  preLIF1: bigint;
+  preLIF2: bigint;
+  preLiquidationOracle: `0x${string}`;
+}) {
+  return {
+    preLltv: preLiquidiationParams.preLltv,
+    preLCF1: preLiquidiationParams.preLCF1,
+    preLCF2: preLiquidiationParams.preLCF2,
+    preLIF1: preLiquidiationParams.preLIF1,
+    preLIF2: preLiquidiationParams.preLIF2,
+    preLiquidationOracle: preLiquidiationParams.preLiquidationOracle as Address,
+  };
+}
```

Comment:
Is this actually doing anything? I think you can just cast `thing as PreLiquidationParams` right?